Sources/Common/Core/DataArray/index.d.ts

```diff
@@ -64,71 +75,194 @@ export interface vtkDataArray extends vtkObject {
+	 */
+	setTuple(idx: number, tuple: Array): void;
+
+	/**
+	 * Get the tuple at the given index.
+	 *
+	 * The recommended use is the following:
+	 * `const x = [];`
+	 * `dataArray.getTuple(idx, x);`
+	 * instead of:
+	 * `const x = dataArray.getTuple(idx);`
+	 *
+	 * @param {Number} idx
+	 * @param {Array} [tupleToFill] (default [])
+	 * @returns {Array}
+	 */
+	getTuple(idx: number, tupleToFill?: Array): Array;
+	
+	/**
+	 * Get the tuples between fromId (inclusive) and toId (exclusive).
+	 *
+	 * If fromId or toId is negative, it refers to a tuple index from the
+	 * end of the underlying typedArray.
+	 * If the range between fromId and toId is invalid, getTuples returns
+	 * and empty TypedArray.
+	 *
+	 * NOTE: Any changes to the returned TypedArray will result in changes to
+	 * this DataArray's underlying TypedArray.
+	 *
+	 * @param {Number} [fromId] (default: 0)
+	 * @param {Number} [toId] (default: publicAPI.getNumberOfTuples())
+	 * @returns {TypedArray}
+	 */
+	getTuples(fromId?: number, toId?: number): TypedArray;
+
+	/**
+	 * Insert the given tuple at the given index.
+	 * NOTE: May resize the data values array. "Safe" version of setTuple.
+	 *
+	 * A typical usage is when `vtkDataArray` is initialized with
+	 * `initialValues = { size: 0, values: new Uint8Array(1000) }`, where
+	 * an empty but pre-allocated array with 1'000 components is created.
+	 * The component values can then be inserted with `insertTuple()` or
+	 * `insertNextTuple()` without requiring new memory allocation until
+	 * the size of 1'000 is exceeded (e.g. after inserting the 250th
+	 * 4-component tuple).
+	 *
+	 * `insertTuple` increases the number of tuples (`getNumberOfTuples()`).
+	 *
+	 * @see insertNextTuple
+	 * @see getNumberOfTuples
 	 *
 	 * @param {Number} idx
-	 * @param {Number[]} tuple
+	 * @param {Array|TypedArray} tuple
+	 * @returns {Number} Index of the inserted tuple
 	 */
-	setTuple(idx: number, tuple: number[]): void;
+	insertTuple(idx: number, tuple: Array|TypedArray): number;
 
 	/**
+	 * Insert tuples starting at the given idx.
 	 *
 	 * @param {Number} idx
-	 * @param {Number[]} [tupleToFill] (default [])
+	 * @param {Array|TypedArray} tuples Tuples to insert
+	 * @returns The index of the last inserted tuple
 	 */
-	getTuple(idx: number, tupleToFill?: number[]): number[];
+	insertTuples(idx: number, tuples: Array|TypedArray): void;
+
+	/**
+	 * Insert the given tuple at the next available slot and return the index of the insertion.
+	 * NOTE: May resize the data values array. "Safe" version of setTuple.
+	 *
+	 * @see insertTuple
+	 *
+	 * @param {Array|TypedArray} tuple
+	 * @returns {Number} Index of the inserted tuple.
+	 */
+	insertNextTuple(tuple: Array|TypedArray): number;
+
+	/**
+	 * Convenience function to insert an array of tuples with insertNextTuple.
+	 * NOTE: tuples.length must be a multiple of `getNumberOfComponents`.
+	 * @param {Array|TypedArray} tuples
+	 */
+	insertNextTuples(tuples: Array|TypedArray): void;
```

Comment:
`* @returns The index of the last inserted tuple`

Reply:
Array of what ?